runtime/compiler/x/codegen/CallSnippet.cpp

```diff
@@ -67,12 +67,44 @@ uint8_t *TR::X86PicDataSnippet::encodeConstantPoolInfo(uint8_t *cursor)
    uintptrj_t cpAddr = (uintptrj_t)_methodSymRef->getOwningMethod(cg()->comp())->constantPool();
    *(uintptrj_t *)cursor = cpAddr;
 
-   if (_thunkAddress)
+   uintptr_t inlinedSiteIndex = (uintptr_t)-1;
+   if (_startOfPicInstruction->getNode() != NULL)
+      inlinedSiteIndex = _startOfPicInstruction->getNode()->getInlinedSiteIndex();
+
+   if (_hasJ2IThunkInPicData)
+      {
+      TR_ASSERT(
+         TR::Compiler->target.is64Bit(),
+         "expecting a 64-bit target for thunk relocations");
+
+      auto info =
+         (TR_RelocationRecordInformation *)comp()->trMemory()->allocateMemory(
+            sizeof (TR_RelocationRecordInformation),
+            heapAlloc);
+
+      int offsetToJ2IVirtualThunk = isInterface() ? 0x22 : 0x18;
```

Comment:
Is there a way these `0x22` and `0x18` numbers could be computed from "cursor" locations rather than these more brittle magic numbers? For example, remembering the cursor of the constant pool address and the cursor of the thunk address as the PICs are laid down, then generating the external relocation afterwards at which point you can just compute the difference?

Reply:
They could be, although that would separate the creation of `TR_J2IVirtualThunkPointer` from `TR_ConstantPool` and `TR_Thunks`, and the offset constants would still be necessary [elsewhere](https://github.com/eclipse/openj9/blob/bada7c8b47cd8ec96825d3b04837e3321dec47e0/runtime/compiler/x/runtime/X86PicBuilder.inc#L89-L103).

Reply:
*sigh* that's why it works better as a separate relocation record. Ok, if it's too much hassle, I suppose you'll have to live with it ;) .